R/util.R

```diff
@@ -42,8 +42,12 @@ serialize_r_to_tensor <- function(data, path) {
 #' @keywords internal
 deserialize_tensor_to_r <- function(path) {
   content <- readLines(path)
+  converter <- switch(
+    content[[1]],
+    "int" = as.integer,
```

Comment:
Turns out our test for this wasn't actually checking the type! See https://github.com/mrc-ide/frogger/pull/68/files#diff-9019d605dcb8ef5d76c288c13a1c7483f042c215ef2e49b4d4f0b932cad296eeR81 Difference between `expect_equal` and expect_identical.